Add -lvchostif to link when using -lEGL on rpi

This is required because libEGL from userland uses sybols
from this library.

lib/libEGL.so.1.0.0                                                                                                                                                                                                                              121: 00000000     0 FUNC    GLOBAL DEFAULT  UND vc_dispmanx_element_add
  1552: 00000000     0 FUNC    GLOBAL DEFAULT  UND vc_dispmanx_element_add

These symbols are provided by libvchostif as seen below

lib/libvchostif.so
   252: 0000b161   192 FUNC    GLOBAL DEFAULT    9 vc_dispmanx_element_add
   809: 0000b161   192 FUNC    GLOBAL DEFAULT    9 vc_dispmanx_element_add

With this explicit link, plugins fail during runtime

(gst-plugin-scanner:571): GStreamer-WARNING **: Failed to load plugin '/usr/lib/gstreamer-1.0/libgstomx.so': Error relocating /usr/lib/libgstgl-1.0.so.0: vc_dispmanx_element_add: symbol not found
(gst-plugin-scanner:571): GStreamer-WARNING **: Failed to load plugin '/usr/lib/gstreamer-1.0/libgstopengl.so': Error relocating /usr/lib/libgstgl-1.0.so.0: vc_dispmanx_element_add: symbol not found

Signed-off-by: Khem Raj <raj.khem@gmail.com>

Upstream-Status: Pending


Move patch from gstreamer1.0-plugins-bad and rebase for gstreamer1.0-plugins-base.

Signed-off-by: Kai Kang <kai.kang@windriver.com>
---
diff --git a/m4/gst-gl.m4 b/m4/gst-gl.m4
index 1e9724094..ab7774c5c 100644
--- a/m4/gst-gl.m4
+++ b/m4/gst-gl.m4
@@ -231,7 +231,7 @@ case $host in
                             HAVE_EGL=yes
                             HAVE_GLES2=yes
                             HAVE_EGL_RPI=yes
-                            EGL_LIBS="-lbcm_host -lvcos -lvchiq_arm"
+                            EGL_LIBS="-lbcm_host -lvchostif -lvcos -lvchiq_arm"
                             EGL_CFLAGS=""
                             AC_DEFINE(USE_EGL_RPI, [1], [Use RPi platform])
                           ])
